src/ProjectTemplates/BlazorWasm.ProjectTemplates/content/BlazorWasm-CSharp/Client/wwwroot/service-worker.published.js

```diff
@@ -0,0 +1,57 @@
+// Caution! Be sure you understand the caveats before publishing an application with
+// offline support. See https://aka.ms/blazor-offline-considerations
+//
+// This service worker implements a cache-first policy. For offline support to work,
+// you must ensure that 'resourcesAvailableOffline' lists all the resources needed
+// by your application (except for .NET assemblies and dotnet.wasm).
+
+const resourcesAvailableOffline = [
+    'index.html',
+    'manifest.json',
+    'icon-512.png',
+    '_framework/blazor.boot.json',
+    '_framework/blazor.webassembly.js',
+    '_framework/wasm/dotnet.js',
+    'css/site.css',
+    'css/bootstrap/bootstrap.min.css',
+    'css/open-iconic/font/css/open-iconic-bootstrap.min.css',
+    'css/open-iconic/font/fonts/open-iconic.woff'
```

Comment:
There are other font files in the same dir (e.g., `open-iconic.ttf`, `open-iconic.otf`, etc.). However they only exist for fallback when used on older browsers, e.g., IE11 with the polyfills needed for Blazor.

On browsers that support service workers, the only font file they will choose is the `.woff` one, so it's the only one we want to include here.

This kind of thing is one of the reasons why it's not desirable to try generating this list automatically. The same would apply to static content from 3rd-party NuGet packages - we have no way to know which of them you really want to use.

Developers who have more complicated requirements may want to generate their `service-worker.js` file using Webpack-based tooling. This is capable of spidering through your resources, starting with `index.html` and discovering chains of references (e.g., `first.css` imports `second.css` which references `my-background.png`). The way we're approaching service workers in this PR allows for the use of arbitrary external tooling like that.

Reply:
OK, following discussions, we've decided to produce an assets manifest similar to what @javiercn describes above. It's now implemented in this PR.